src/App.js

```diff
@@ -1,60 +1,91 @@
-      this.setState((state) => ({
-        // Store message key so we can use it as a key in our list items when rendering messages
-        messages: [...state.messages, { key: data.key, val: data.val() }],
-      }));
+const theme = createTheme({
+  palette: {
+    primary: indigo,
+    secondary: teal,
+  },
+});
+const DB_POSTS_KEY = "posts";
+
+export default function App() {
+  const [posts, setPosts] = useState([]);
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    onAuthStateChanged(auth, (user) => {
+      setUser(user);
+    });
+
+    const postsRef = ref(database, DB_POSTS_KEY);
+
+    onChildAdded(postsRef, (data) => {
+      setPosts((prevPosts) => [
+        ...prevPosts,
+        { key: data.key, val: data.val() },
+      ]);
     });
-  }
 
-  // Note use of array fields syntax to avoid having to manually bind this method to the class
-  writeData = () => {
-    const messageListRef = ref(database, DB_MESSAGES_KEY);
-    const newMessageRef = push(messageListRef);
-    set(newMessageRef, "abc");
+    onChildChanged(postsRef, (data) => {
+      setPosts((prevPosts) => {
+        const replacePostIndex = prevPosts.findIndex(
+          (post) => post.key === data.key
+        );
+        const replace = prevPosts.slice();
+        replace[replacePostIndex] = { key: data.key, val: data.val() };
+        return replace;
+      });
+    });
+  }, []);
+
+  const updateUser = () => {
+    setUser({ ...auth.currentUser });
   };
 
-  render() {
-    // Convert messages in state to message JSX elements to render
-    let messageListItems = this.state.messages.map((message) => (
-      <li key={message.key}>{message.val}</li>
-    ));
+  const commentRoute = posts.map((post) => {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-          {/* TODO: Add input field and add text input as messages in Firebase */}
-          <button onClick={this.writeData}>Send</button>
-          <ol>{messageListItems}</ol>
-        </header>
-      </div>
+      <Route
+        path={`${post.key}`}
+        element={<Comment post={post} user={user} />}
+      />
     );
-  }
+  });
+  console.log(user);
+  return (
+    <div className="App">
+      <div className="App-header">
+        <ThemeProvider theme={theme}>
+          <BrowserRouter>
+            <UserBar user={user} updateUser={updateUser} />
+            <Routes>
+              <Route
+                path="/"
+                element={<MainPage posts={posts} user={user} />}
+              />
+              <Route path="/logIn" element={<LogInForm />} />
+              <Route path="/signUp" element={<SignUpForm />} />
+              <Route path="/comment" element={<Post posts={posts} />}>
+                {commentRoute}
+              </Route>
```

Comment:
```suggestion
              <Route path="/comment/:id" element={<Post posts={posts} />
```

instead of creating a single comment component for each post and placing it as child here, it probably would make more sense to get a dynamic path parameter here and then render accordingly.